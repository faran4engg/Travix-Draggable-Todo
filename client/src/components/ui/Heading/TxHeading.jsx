import React from "react";
import styled from "styled-components";
import BillabongFont from "../../../assets/font/Billabong.woff";

import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const H1 = styled.h1`
  font-family: BillabongFont, "BillabongFont";
  text-align: center;
  font-weight: 100;
  font-size: 4rem;
  word-spacing: 0.5rem;

  color: ${props => (props.color ? props.color : "#ccc")};
  text-shadow: 0px 4px 0 black;

 &:hover{
  color: #e4e1e1;
  text-shadow: 0px 4px 0 ${props => (props.color ? props.color : "#ccc")};

 
 }

  @font-face {
    font-family: BillabongFont;
    font-style: normal;
    font-weight: normal;
    src: url( '${BillabongFont}') format("woff");
  }
`;

const TxHeading = ({ content, color, path }) => {
  return path ? (
    <Link to={path}>
      <H1 color={color}>{content}</H1>
    </Link>
  ) : (
    <H1 color={color}>{content}</H1>
  );
};

TxHeading.propTypes = {
  content: PropTypes.string,
  color: PropTypes.string,
  path: PropTypes.string
};

export default TxHeading;
