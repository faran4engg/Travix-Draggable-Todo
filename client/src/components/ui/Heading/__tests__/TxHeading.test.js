import React from "react";

import { shallow, mount } from "enzyme";
import TxHeading from "../TxHeading";
describe("TxHeading", () => {
  it("should render correctly ", () => {
    const component = shallow(<TxHeading content="Travix" color="#80e0f6" />);

    expect(component).toMatchSnapshot();
  });

  describe("when passing props", () => {
    it("should render content correctly ", () => {
      const component = mount(<TxHeading content="Travix" color="#80e0f6" />);
      expect(component.find("h1").text()).toEqual("Travix");
    });
  });
});
