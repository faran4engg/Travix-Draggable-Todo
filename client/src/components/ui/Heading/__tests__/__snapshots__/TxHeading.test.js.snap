// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxHeading should render correctly  1`] = `ShallowWrapper {}`;
